qiskit/providers/backend_compat.py

```diff
@@ -13,188 +13,324 @@
+from qiskit.providers.models.backendproperties import Gate as GateSchema
+
+logger = logging.getLogger(__name__)
 
 
 def convert_to_target(
     configuration: BackendConfiguration,
-    properties: BackendProperties = None,
-    defaults: PulseDefaults = None,
+    properties: Optional[Union[BackendProperties, Dict]] = None,
+    defaults: Optional[Union[PulseDefaults, Dict]] = None,
     custom_name_mapping: Optional[Dict[str, Any]] = None,
-    add_delay: bool = False,
-    filter_faulty: bool = False,
+    add_delay: bool = True,
+    filter_faulty: bool = True,
 ):
-    """Uses configuration, properties and pulse defaults
-    to construct and return Target class.
+    """Decode transpiler target from backend data set.
+
+    This function directly generates ``Target`` instance without generating
+    intermediate legacy objects such as ``BackendProperties`` and ``PulseDefaults``.
+
+    .. note::
+        Passing in legacy objects like BackendProperties as properties and PulseDefaults
+        as defaults will be deprecated in the future.
+
+    Args:
+        configuration: Backend configuration as ``BackendConfiguration``
+        properties: Backend property dictionary or ``BackendProperties``
+        defaults: Backend pulse defaults dictionary or ``PulseDefaults``
+        custom_name_mapping: A name mapping must be supplied for the operation
+        not included in Qiskit Standard Gate name mapping, otherwise the operation
+        will be dropped in the resulting ``Target`` object.
+        add_delay: If True, adds delay to the instruction set.
+        filter_faulty: If True, this filters the non-operational qubits.
+
+    Returns:
+        A ``Target`` instance.
+    """
 
-    In order to convert with a ``defaults.instruction_schedule_map``,
-    which has a custom calibration for an operation,
-    the operation name must be in ``configuration.basis_gates`` and
-    ``custom_name_mapping`` must be supplied for the operation.
-    Otherwise, the operation will be dropped in the resulting ``Target`` object.
+    # importing pacakges where they are needed, to avoid cyclic-import.
+    from qiskit.transpiler.target import Target  # pylint: disable=cyclic-import
+    from qiskit.circuit.controlflow import ForLoopOp, IfElseOp, SwitchCaseOp, WhileLoopOp
+    from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
+    from qiskit.qobj.pulse_qobj import PulseLibraryItem
+    from qiskit.qobj.converters.pulse_instruction import QobjToInstructionConverter
+    from qiskit.providers.models.pulsedefaults import Command
+    from qiskit.pulse.calibration_entries import PulseQobjDef
+    from qiskit.circuit.parameter import Parameter
+    from qiskit.circuit.gate import Gate
+
+    required = ["measure", "delay"]
+    if isinstance(defaults, PulseDefaults):
+        defaults = defaults.to_dict()
+
+    if isinstance(properties, BackendProperties):
+        properties = properties.to_dict()
```

Comment:
```suggestion
    required = ["measure", "delay"]
    if isinstance(defaults, PulseDefaults):
        defaults = defaults.to_dict()

    if isinstance(properties, BackendProperties):
        properties = properties.to_dict()
```
It's preferable not to convert Qiskit object back into dictionary. The IBM Provider implementation assumes conversion of the legacy JSON _data source_ into the V2 model, while this code is for the conversion of the legacy Qiskit BackendV1 object into the V2 model. Since backend properties and defaults are also legacy objects which are not expected to appear in the V2 model, the original code bypasses creation of these objects for performance. However, in the situation where this code is executed, you already have a complete V1 backend object with properties and defaults objects. So converting them back into dict and re-applying schema is just an overhead. You need to rewrite the rest of code to use attribute instead of dict getitem, but you can reuse the machinery :)

Reply:
Applied suggestion :)